BinaryObjectScanner/Protection/CopyX.cs

```diff
@@ -0,0 +1,171 @@
+        // TODO: Check the last directory alphabetically and not just ZDAT*
+
+        /// <inheritdoc/>
+        public string? CheckPortableExecutable(string file, PortableExecutable pex, bool includeDebug)
+        {
+            // Checks for Professional
+            if (pex.OverlayStrings != null)
+            {
+                // Checks if main executable contains reference to optgraph.dll. Emergency 4's is missing this for some reason. 
+                // This might be better removed later, as Redump ID 82475 is a false positive, and also doesn't actually contain the actual optgraph.dll file.
+                // TODO: Find a way to check for situations like Redump ID 48393, where the string is spaced out with 0x00 between letters and does not show up on string checks.
+                // TODO: This might need to check every single section. Unsure until more samples are acquired.
+                if (pex.OverlayStrings.Any(s => s.Contains("optgraph.dll")))
+                {
+                    // TODO: TKKG also has an NE 3.1x executable with a reference. This can be added later.
+                    // Samples: Redump ID 108150
+                    return "copy-X";
+                }
+            }
+
+            var strs = pex.GetFirstSectionStrings(".rdata");
+            if (strs != null)
+            {
+                if (strs.Any(s => s.Contains("optgraph.dll")))
+                {
+                    // Samples: Redump ID 82475 (False positive, but probably catches original Emergency 2), would catch Redump ID 48393
+                    return "copy-X";
+                }
+            }
+
+            return null;
+        }
+
+        /// <inheritdoc/>
+#if NET20 || NET35
+        public Queue<string> CheckDirectoryPath(string path, IEnumerable<string>? files)
+#else
+        public ConcurrentQueue<string> CheckDirectoryPath(string path, IEnumerable<string>? files)
+#endif
+        {
+#if NET20 || NET35
+            var protections = new Queue<string>();
+#else
+            var protections = new ConcurrentQueue<string>();
+#endif
+            // Checks for Light
+
+            if (files == null)
+                return protections;
+            //Gets files in ZDAT*
+            var zdatFiles = files.Where(f => f.Remove(0, path.Length + 1).StartsWith("ZDAT", StringComparison.OrdinalIgnoreCase));
+            var fileList = zdatFiles.ToList();
+            // Sorts list of files in ZDAT* so I can just pull the first one, later ones have a chance of the ring intersecting the start of the file.
+            fileList.Sort();
+            if (fileList.Count > 0)
+            {
+                try
+                {
+                    FileStream stream = new FileStream(fileList[0], FileMode.Open, FileAccess.Read);
+                    byte[] block = new byte[1024];
+                    stream.Read(block, 0, 1024);
```

Comment:
These two lines can be combined because of the `SabreTools.IO.Extensions`. I have a helper method that allows you to call `stream.ReadBytes(XXX)` to do basically this.

Reply:
Please add a newline after this block of code.